### PR TITLE
[CARBONDATA-1766] fix serialization issue for CarbonAppendableStreamSink

### PR DESCRIPTION
**Pull request by QiangCai**
fix serialization issue for CarbonAppendableStreamSink.

 - [ x] Any interfaces changed?
 no
 - [x] Any backward compatibility impacted?
 no
 - [x] Document update required?
 no
 - [ x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1272/

**Comment by jackylk** [comment]
LGTM

File: hadoop/src/main/java/org/apache/carbondata/hadoop/streaming/CarbonStreamRecordWriter.java [hunk 1]
```diff
@@ -99,13 +100,21 @@
...
     // set basic information
     hadoopConf = job.getConfiguration();
-    CarbonLoadModel carbonLoadModel = CarbonStreamOutputFormat.getCarbonLoadModel(hadoopConf);
     if (carbonLoadModel == null) {
-      throw new IOException(
-          "CarbonStreamRecordWriter require configuration: mapreduce.output.carbon.load.model");
+      carbonLoadModel = CarbonStreamOutputFormat.getCarbonLoadModel(hadoopConf);
+      if (carbonLoadModel == null) {
+        throw new IOException(
+            "CarbonStreamRecordWriter require configuration: mapreduce.output.carbon.load.model");
```
Comment (jackylk) [inline hadoop/src/main/java/org/apache/carbondata/hadoop/streaming/CarbonStreamRecordWriter.java]: This configuration should start with `carbon`, change to `carbon.load.model`

Comment (jackylk) [reply hadoop/src/main/java/org/apache/carbondata/hadoop/streaming/CarbonStreamRecordWriter.java]: Do you really require the whole CarbonLoadModel? It seems you only need CarbonTable object, right?

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**